app/main/views/login.py

```diff
@@ -1,32 +1,56 @@
-from flask import request, render_template, url_for, redirect, session
+from flask import request, render_template, url_for, redirect, \
+    current_app, flash
+from flask_login import login_user, logout_user, login_required
+from dmutils.user import user_has_role, User
 
 from .. import main
-from ..helpers.auth import check_auth
+from ... import data_api_client
+from ..forms import LoginForm
 from . import get_template_data
 
 
-@main.route('/login', methods=['POST', 'GET'])
-def login():
-    if request.method == 'GET':
-        return render_template("login.html", **get_template_data({
-            "previous_responses": None,
-            "logged_out": "logged_out" in request.args
-        }))
-    if check_auth(
-        request.form['username'],
-        request.form['password'],
-        main.config['PASSWORD_HASH']
-    ):
-        session['username'] = request.form['username']
-        return redirect(url_for('.index'))
+@main.route('/login', methods=['GET'])
+def render_login():
+    form = LoginForm()
+    return render_template('login.html', **get_template_data(form=form))
+
+
+@main.route('/login', methods=['POST'])
+def process_login():
+    next_url = request.args.get('next')
+    form = LoginForm()
+    if form.validate_on_submit():
+        user_json = data_api_client.authenticate_user(
+            form.email_address.data,
+            form.password.data,
+            supplier=False)
+
+        if not user_has_role(user_json, 'admin'):
+            message = "login.fail: Failed to log in: %s"
+            current_app.logger.info(message, form.email_address.data)
+            flash('no_account', 'error')
+            return render_template(
+                'login.html',
+                **get_template_data(form=form, next=next_url)
+            ), 403
 
-    return render_template("login.html", **get_template_data({
-        "error": "Could not log in",
-        "previous_responses": request.form
-    }))
+        user = User.from_json(user_json)
+        login_user(user)
+
+        if next_url and next_url.startswith('/admin'):
```

Comment:
This doesn't seem to work - it just takes me to the "Find a service" page no matter what is in as `next`.